cmpserver/plugin/plugin.go

```diff
@@ -32,14 +43,78 @@ func NewService(initConstants CMPServerInitConstants) *Service {
 	}
 }
 
-func runCommand(command Command, path string, env []string) (string, error) {
+func runCommand(ctx context.Context, command Command, path string, env []string) (string, error) {
 	if len(command.Command) == 0 {
 		return "", fmt.Errorf("Command is empty")
 	}
-	cmd := exec.Command(command.Command[0], append(command.Command[1:], command.Args...)...)
+	cmd := exec.CommandContext(ctx, command.Command[0], append(command.Command[1:], command.Args...)...)
+
 	cmd.Env = env
 	cmd.Dir = path
-	return executil.Run(cmd)
+
+	execId, err := rand.RandString(5)
```

Comment:
I pulled all this from `pkg` rather than putting up a PR and making that code more convoluted.

The error stuff could probably be simplified here, but I'm trying to match current behavior as closely as possible since I don't know the background on why the `pkg` code was written the way it was.